server/bundler/loader.js

```diff
@@ -68,30 +68,31 @@ function splitTemplate( path, module, chunkName ) {
 
 	result = [
 		'page( ' + path + ', function( context, next ) {',
-		'	if ( _loadedSections[ ' + JSON.stringify( module ) + ' ] ) {',
+		'	if ( _loadedSections[ ' + JSON.stringify( section.module ) + ' ] ) {',
+		'		context.store.dispatch( { type: "SET_SECTION", section: ' + JSON.stringify( section ) + ' } );',
 		'		layoutFocus.next();',
 		'		return next();',
 		'	}',
 		'	context.store.dispatch( { type: "SET_SECTION", isLoading: true } );',
-		'	context.store.dispatch( { type: "SET_SECTION", chunkName: ' + JSON.stringify( chunkName ) + ' } );',
 		'	require.ensure([], function( require, error ) {',
 		'		if ( error ) {',
 		'			if ( ! LoadingError.isRetry() ) {',
-		'				LoadingError.retry( ' + JSON.stringify( chunkName ) + ' );',
+		'				LoadingError.retry( ' + JSON.stringify( section.name ) + ' );',
 		'			} else {',
 		'				context.store.dispatch( { type: "SET_SECTION", isLoading: false } );',
-		'				LoadingError.show( ' + JSON.stringify( chunkName ) + ' );',
+		'				LoadingError.show( ' + JSON.stringify( section.name ) + ' );',
 		'			}',
 		'			return;',
 		'		}',
 		'		context.store.dispatch( { type: "SET_SECTION", isLoading: false } );',
-		'		if ( ! _loadedSections[ ' + JSON.stringify( module ) + ' ] ) {',
-		'			require( ' + JSON.stringify( module ) + ' )( controller.clientRouter );',
-		'			_loadedSections[ ' + JSON.stringify( module ) + ' ] = true;',
+		'		context.store.dispatch( { type: "SET_SECTION", section: ' + JSON.stringify( section ) + ' } );',
```

Comment:
Can we merge this `SET_SECTION` dispatch and the one above (the `isLoading` one)?